packages/qwik-city/middleware/request-handler/page-handler.ts

```diff
@@ -21,6 +26,8 @@ export function pageHandler<T = any>(
   const { status, headers } = userResponse;
   const { response } = requestCtx;
   const isPageData = userResponse.type === 'pagedata';
+  const requestHeaders: Record<string, string> = {};
+  requestCtx.request.headers.forEach((value, key) => (requestHeaders[key] = value));
```

Comment:
We have something similar in our React codebase, but we allow registering functions that extra environment data from a request object.

This way, more than headers can be used to set the environment data, it avoids having to copy all headers, and it allows running the render without a request object, like when you're rendering a PDF.